snapcraft/plugins/v2/_plugin.py

```diff
@@ -66,3 +66,13 @@ def get_build_commands(self) -> List[str]:
         snapcraftctl can be used in the script to call out to snapcraft
         specific functionality.
         """
+
+    @property
+    def out_of_source_build(self) -> bool:
```

Comment:
Why not just have this be an attribute?

Reply:
I had originally marked this with `@abc.abstractmethod`, but removed it so I could have the other plugins take a default implementation.

I suppose it would make sense to either switch it to a regular attribute, or add back the `abstractmethod` decorator which would ensure every plugin made a choice.

Reply:
+1 for class attribute with default False like you have here